### ISSUE TITLE
generation stops suddenly

### ISSUE THREAD

**Issue by KillyTheNetTerminal** [patientx/ComfyUI-Zluda#11]
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/76c8b6b0-7d45-4bba-8641-a01c52e4b6ba)

**Comment by patientx** [comment]
please post latest log.

**Comment by KillyTheNetTerminal** [comment]
## ComfyUI-Manager: installing dependencies done.
[2024-06-17 09:42] ** ComfyUI startup time: 2024-06-17 09:42:11.490890
[2024-06-17 09:42] ** Platform: Windows
[2024-06-17 09:42] ** Python version: 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
[2024-06-17 09:42] ** Python executable: E:\ComfyUI-Zluda\venv\Scripts\python.exe
[2024-06-17 09:42] ** Log path: E:\ComfyUI-Zluda\comfyui.log
[2024-06-17 09:42] 
Prestartup times for custom nodes:
[2024-06-17 09:42]    1.5 seconds: E:\ComfyUI-Zluda\custom_nodes\ComfyUI-Manager
[2024-06-17 09:42] 
Total VRAM 8192 MB, total RAM 32688 MB
[2024-06-17 09:42] pytorch version: 2.2.1+cu118
[2024-06-17 09:42] Set vram state to: NORMAL_VRAM
[2024-06-17 09:42] ***--------------------------------ZLUDA------------------------------------***
[2024-06-17 09:42] Detected ZLUDA, support for it is experimental and comfy may not work properly.
[2024-06-17 09:42] Disabling cuDNN because ZLUDA does currently not support it.
[2024-06-17 09:42] Disabling flash because ZLUDA does currently not support it.
[2024-06-17 09:42] Enabling math_sdp.
[2024-06-17 09:42] Disabling mem_efficient_sdp because ZLUDA does currently not support it.
[2024-06-17 09:42] ***-------------------------------------------------------------------------***
[2024-06-17 09:42] Device: cuda:0 AMD Radeon RX 580 2048SP [ZLUDA] : native
[2024-06-17 09:42] Using pytorch cross attention
[2024-06-17 09:42] ### Loading: ComfyUI-Impact-Pack (V5.13.2)
[2024-06-17 09:42] ### Loading: ComfyUI-Impact-Pack (Subpack: V0.6)
[2024-06-17 09:42] ### Loading: ComfyUI-Manager (V2.38.1)
[2024-06-17 09:42] [Impact Pack] Wildcards loading done.
[2024-06-17 09:42] ### ComfyUI Revision: 2350 [a117ea41] | Released on '2024-06-17'
[2024-06-17 09:42] 
Import times for custom nodes:
[2024-06-17 09:42]    0.0 seconds: E:\ComfyUI-Zluda\custom_nodes\ComfyUI-deepcache
[2024-06-17 09:42]    0.0 seconds: E:\ComfyUI-Zluda\custom_nodes\websocket_image_save.py
[2024-06-17 09:42]    0.5 seconds: E:\ComfyUI-Zluda\custom_nodes\ComfyUI-Manager
[2024-06-17 09:42]    1.3 seconds: E:\ComfyUI-Zluda\custom_nodes\ComfyUI-Impact-Pack
[2024-06-17 09:42] 
[2024-06-17 09:42] Starting server

[2024-06-17 09:42] To see the GUI go to: http://127.0.0.1:8188
[2024-06-17 09:42] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/model-list.json
[2024-06-17 09:42] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
[2024-06-17 09:42] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/github-stats.json
[2024-06-17 09:42] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/custom-node-list.json
[2024-06-17 09:42] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
[2024-06-17 09:42] got prompt
[2024-06-17 09:42] model_type EPS
[2024-06-17 09:42] Using pytorch attention in VAE
[2024-06-17 09:42] Using pytorch attention in VAE
[2024-06-17 09:42] Requested to load SD1ClipModel
[2024-06-17 09:42] Loading 1 new model
[2024-06-17 09:42] E:\ComfyUI-Zluda\comfy\ldm\modules\attention.py:407: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
  out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)
[2024-06-17 09:42] Requested to load BaseModel
[2024-06-17 09:42] Loading 1 new model
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/70d698df-8d31-4d6a-a8eb-50634e0414b7)


(

**Comment by KillyTheNetTerminal** [comment]
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/ea883dcf-c0ce-4361-bc6c-999217523ba8)
with --lowvram arg

**Comment by patientx** [comment]
you were doing ok yesterday what happened ? it was working at least with sd1.5 , wasnt it ?

**Comment by KillyTheNetTerminal** [comment]
I think the only thing different I did yesterday was eliminate the extensions that came by default and only leave Deepcache. That error also occurred, I don't know, after insisting a few times I did it but it turned out to be unstable, this is a fresh installation because I wanted to try to get the 3D pack that mostly uses Cuda and other things not compatible with directml to work

I saw that there is a reference to flash attention? https://github.com/lks-ai/ComfyUI-StableAudioSampler/issues/2#issuecomment-2173439575

There are many things that I don't understand but I associate it with this: :\ComfyUI-Zluda\comfy\ldm\modules\attention.py:407: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
 out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)

**Comment by patientx** [comment]
ignore that, that's normal error

**Comment by KillyTheNetTerminal** [comment]
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/b239f8d6-84ef-4b21-afb8-e536f094f798)
I have the HIP SDK installed, replace the files in the Library folder and create the environment variable
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/920d2c67-cbe9-448b-bb24-fefbac8dcab2)

**Comment by patientx** [comment]
but know this : rx 580 has some kind of problem with zluda regarding vram -in turn this effects the generation since we do it on vram- and ... since it is an "old" gpu , it is wonder to even get it working at all ... regardless you can try sdnext but you probably would have similar problems there -in fact I know about 580's problems from them-.

Is your python from windows store or from its website ?

**Comment by KillyTheNetTerminal** [comment]
Yes, I really think it seems like a problem with memory management. Since that seems to be the way it reacts when we run out of Vram memory, although it is in lowvram. My Python is not the one in the store

**Comment by patientx** [comment]
there should be an older sdnext version which uses those older torch versions, revert to that and try it but these memory problems popup there too unfortunately...

**Comment by KillyTheNetTerminal** [comment]
I understand, thank you very much. Do you think there is any way to solve something like this? I wonder why the other installation failed less

**Comment by KillyTheNetTerminal** [comment]
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/2d73453a-d6a7-4737-b7f6-9c39cd7488ce)
after deleting this cache file
![image](https://github.com/patientx/ComfyUI-Zluda/assets/120438550/2b9a93a2-6b8d-4391-8060-c11bd1763b12)

**Comment by patientx** [comment]
so you rebuild the cache, well done.

**Comment by KillyTheNetTerminal** [comment]
Your explanation is a bit beyond my understanding but I guess so :)
